File: C++/Items.cpp
#include <bits/stdc++.h>

using namespace std;

class Items
{

    private :

        string id;
        string nama_items;
        int harga;

    public :

    Items(){}

    Items(string id, string nama, int harga)
    {
        this->id = id;
        this->nama_items = nama;
        this->harga = harga;
    }

    // Setter
    void setId(string id) 
    {
        this->id = id;
    }

    void setNamaItems(string nama_items) 
    {
        this->nama_items = nama_items;
    }

    void setHarga(int harga) 
    {
        this->harga = harga;
    }

    // Getter
    string getId() 
    {
        return this->id;
    }

    string getNamaItems() 
    {
        return this->nama_items;
    }

    int getHarga() 
    {
        return this->harga;
    }

    int id_length()
    {
        return id.length();
    }

    int name_length()
    {
        return nama_items.length();
    }

    int harga_length()
    {
        return to_string(harga).length();
    }

    ~Items(){}
    
};

File: Java/Items.java
public class Items 
{

    // atribut private
    private String id;
    private String namaItems;
    private int harga;

    // constructor default
    public Items() {}

    // constructor dengan parameter
    public Items(String id, String nama, int harga) 
    {
        this.id = id;
        this.namaItems = nama;
        this.harga = harga;
    }

    // Setter
    public void setId(String id) 
    {
        this.id = id;
    }

    public void setNamaItems(String namaItems) 
    {
        this.namaItems = namaItems;
    }

    public void setHarga(int harga) 
    {
        this.harga = harga;
    }

    // Getter
    public String getId() 
    {
        return this.id;
    }

    public String getNamaItems() 
    {
        return this.namaItems;
    }

    public int getHarga() 
    {
        return this.harga;
    }

    // Method tambahan
    public int idLength() 
    {
        return id != null ? id.length() : 0;
    }

    public int nameLength() 
    {
        return namaItems != null ? namaItems.length() : 0;
    }

    public int hargaLength() 
    {
        return String.valueOf(harga).length();
    }
}
